Fix duplicated imports

c-bata opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/guide/pull-requests.md#the-pull-request-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
5. If this PR changes image versions, please title this PR "Bump <image name> from x.x.x to y.y.y."
-->

**What this PR does / why we need it**:

* Add `__init__.py` to avoid using namespace package.
    * See https://dev.to/methane/don-t-omit-init-py-3hga.
* Fix duplicated import problems by `__all__` attribute.
    * To clarify this problem, I attached a screenshot of PyCharm.
    * ![ScreenShot 2020-04-09 23 15 58](https://user-images.githubusercontent.com/5564044/78904830-18fc9900-7ab8-11ea-833c-fb9a018cac9c.png)
    * `logging` is imported at L3 and used at L8. But Jetbrains IDE highlights `import logging` as gray. This means it is an unused line. The reason why it is highlighted as gray is using star imports ( `from pkg.suggestion.v1alpha3.internal.search_space import *` and `from pkg.suggestion.v1alpha3.internal.trial import *` ). ~To avoid this problem, I define ``__all__`` attribute. See https://docs.python.org/3/tutorial/modules.html#importing-from-a-package~

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
None

**Special notes for your reviewer**:

None

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

andreyvelich commented on pkg/suggestion/v1alpha3/chocolate/base_chocolate_service.py:
```diff
@@ -1,11 +1,8 @@
 
-from pkg.suggestion.v1alpha3.internal.search_space import *
-from pkg.suggestion.v1alpha3.internal.trial import *
+from pkg.suggestion.v1alpha3.internal import *
```
In some files we are using directly functions/variables import like here: https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1alpha3/chocolate_service.py#L8, but here we are using `import *`.
Maybe in all python files we should avoid `import *` and use only relevant imports, so we can easily see which method we are using from `search_space` and `trial` file?

What do you think?

c-bata replied:
> Maybe in all python files we should avoid import * and use only relevant imports, so we can easily see which method we are using from search_space and trial file?

Yes, I agree you opinion. star-imports are known as anti-pattrens in Python.
OK, I'll remove star-imports.

c-bata commented:
It seems all star-imports are removed.

```
(venv) $ git co master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
(venv) $ git grep "import \*" | grep "\.py"
pkg/suggestion/v1alpha3/chocolate/base_chocolate_service.py:from pkg.suggestion.v1alpha3.internal.search_space import *
pkg/suggestion/v1alpha3/chocolate/base_chocolate_service.py:from pkg.suggestion.v1alpha3.internal.trial import *
pkg/suggestion/v1alpha3/hyperopt/base_hyperopt_service.py:from pkg.suggestion.v1alpha3.internal.search_space import *
pkg/suggestion/v1alpha3/hyperopt/base_hyperopt_service.py:from pkg.suggestion.v1alpha3.internal.trial import *
pkg/suggestion/v1alpha3/skopt/base_skopt_service.py:from pkg.suggestion.v1alpha3.internal.search_space import *
pkg/suggestion/v1alpha3/skopt/base_skopt_service.py:from pkg.suggestion.v1alpha3.internal.trial import *
(venv) $ git co fix-duplicated-imports
Switched to branch 'fix-duplicated-imports'
(venv) $ git grep "import \*" | grep "\.py"
(venv) $ 
```

johnugeorge commented on pkg/suggestion/v1alpha3/chocolate/base_chocolate_service.py:
```diff
@@ -1,11 +1,11 @@
-from pkg.suggestion.v1alpha3.internal.search_space import *
-from pkg.suggestion.v1alpha3.internal.trial import *
+from pkg.suggestion.v1alpha3.internal.search_space import (
+    INTEGER, DOUBLE, CATEGORICAL, DISCRETE, MAX_GOAL,
```
Can we do this in a better way to look more clean? we might have more constants. May be wrap this?

c-bata replied:
You mean that you want to import all constants, right?
Another solution is to define `__all__` attribute like [this revision](https://github.com/kubeflow/katib/commit/f99736db4c79009e8988b498090116b5b09230bb). But as the discussion of https://github.com/kubeflow/katib/pull/1133#discussion_r406294910, it is known as an anti-pattern in Python.

andreyvelich replied:
@c-bata What do you think about creating file: `constant.py` under `/internal/` folder ?
And insert all constant there? I think it will be easier to have all constant in one file

c-bata replied:
It sounds good 👍  @johnugeorge What do you think?

johnugeorge replied:
Yes. SGTM

johnugeorge commented:
/lgtm
/approve
